symmetric/src/hasher.rs

```diff
@@ -1,8 +1,26 @@
+/// A generic trait for cryptographic hashers that consume an arbitrary sequence of input items
+/// and produce a fixed-size output.
+///
+/// This trait abstracts over hash functions in a flexible way, supporting both field elements,
+/// scalars, or any other data type that implements `Clone`.
 pub trait CryptographicHasher<Item: Clone, Out>: Clone {
+    /// Hash an iterator of input items into a single digest output.    ///
```

Comment:
Probably best to either remove ` into a single digest output.` or add it to all the doc comments.
```suggestion
    /// Hash an iterator of input items.
```